examples/host_func_test.go

```diff
@@ -37,8 +44,8 @@ func Test_hostFunc(t *testing.T) {
 		bufAddr := ret[0]
 
 		// Store the address info to the memory.
-		binary.LittleEndian.PutUint32(ctx.Memory.Buffer[retBufPtr:], uint32(bufAddr))
-		binary.LittleEndian.PutUint32(ctx.Memory.Buffer[retBufSize:], bufferSize)
+		ctx.Memory.PutUint32(retBufPtr, uint32(bufAddr))
```

Comment:
not for this PR, but this reminds me we are overdue making things interfaces as direct access to buffer like this is easy to get wrong! https://github.com/tetratelabs/wazero/issues/204